test/codeu/chat/server/DerbyStoreTest.java

```diff
@@ -0,0 +1,190 @@
+				conversation.title.equals(testConversation.title));
+		
+		assertTrue(
+				"Check that the conversation has the same userInvolved when saved",
+				conversation.users.equals(testConversation.users));
+		
+	}
+	
+	@Test
+	public void updateMessagesForConversation() {
+		final User user = controller.newUser("user");
+		final Conversation conversation = controller.newConversation("TEST CONVERSATION", user.id);
+		final Message firstMessage = controller.newMessage(user.id, conversation.id, "TEST FIRST MESSAGE");
+		final Message lastMessage = controller.newMessage(user.id, conversation.id, "TEST LAST MESSAGE");
+		Conversation testConversation = null;
+
+		try {
+			 //ds.addConversation(conversation);
+			 conversation.firstMessage = firstMessage.id;
+			 conversation.lastMessage = lastMessage.id;
+			 ds.updateConversationMessages(conversation);
+			 testConversation = ds.getConversation(conversation.id);
+		 }
+		 catch (Exception ex) {
+			 ex.printStackTrace();
+		 }
+
+		assertTrue(
+				 "Check that messages can be updated once a conversation is made in the conversation table",
+				 Uuid.equals(testConversation.firstMessage, firstMessage.id) && Uuid.equals(testConversation.lastMessage, lastMessage.id));
+		
+	}
+	
+	@Test
+	public void getMessage() {
+		final User user = controller.newUser("user");
+		final Conversation conversation = controller.newConversation("TEST CONVERSATION", user.id);
+		final Message message = controller.newMessage(user.id, conversation.id, "TEST FIRST MESSAGE");
+		Message testMessage = null;
+		
+		try {
+			//ds.addMessage(message);
+			testMessage = ds.getMessage(message.id);
+		}
+		catch (Exception ex) {
+			ex.printStackTrace();
+		}
+		
+		assertTrue(
+				 "Check that messages can be retrieved and have the same id",
+				 Uuid.equals(message.id, testMessage.id));
+		assertTrue(
+				 "Check that messages can be retrieved",
+				 Uuid.equals(message.author, testMessage.author));
+		assertTrue(
+				 "Check that messages can be retrieved",
+				 message.creation.compareTo(message.creation) == 0);
+
+	}
+	
+	@Test
+	public void changeNextMessage() {
+		final User user = controller.newUser("user");
+		final Conversation conversation = controller.newConversation("TEST CONVERSATION", user.id);
+		final Message message = controller.newMessage(user.id, conversation.id, "TEST FIRST MESSAGE");
+		final Message nextMessage = controller.newMessage(user.id, conversation.id, "TEST NEXt MESSAGE");
+		
+		try {
+			//ds.addMessage(message);
+			message.next = nextMessage.id;
+			ds.updateNextMessage(message);
+		}
+		catch (Exception ex) {
+			ex.printStackTrace();
+		}
+		
+		assertTrue(
+				 "Check that messages can be retrieved and have the same id",
+				 Uuid.equals(message.next, nextMessage.id));
+		
+	}
+	
+	
+	// TODO: Add a way to get messages based on conversations and test it
+	// 		 Add a method to update the users involved for a conversation
+	
+	
```

Comment:
You probably want to use a different database for testing than you do for your actual program, and you probably want to delete the test database after each test.